web/src/components/signup-form.js

```diff
@@ -1,16 +1,118 @@
-import React, {Component} from 'react';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import {GoogleLogin} from "react-google-login";
+import {gql} from 'apollo-boost';
+import {Mutation} from 'react-apollo';
+import {withRouter} from 'react-router-dom'
 
 class SignupForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      firstName: '',
+      lastName: '',
+      email: '',
+      // TODO : this will be used when facebook login is added
+      provider: '',
+    };
+  }
+
+  responseGoogle = res => {
+    // Handle google log-in success
+    this.setState({
+      id: res.profileObj.googleId,
+      firstName: res.profileObj.givenName,
+      lastName: res.profileObj.familyName,
+      email: res.profileObj.email,
+      provider: res.tokenObj.idpId
+    });
+  };
+
+  responseFail = err => {
+    // TODO : handle google log-in failure
+    console.error(err);
+  };
+
   render() {
+    // This api client id is ok to be exposed
+    const googleAuthAPIClientID =
+      "420478568442-ltur9qc3g9uam6f166k1pgsa7f2evl5e.apps.googleusercontent.com";
+
+    // Graphql mutation statement for signup
+    const SIGNUP_MUTATION = gql`
+    mutation SignUp($googleId: String!, $email: String!, $firstName: String!, $lastName: String!, $profileImgUrl: String) {
```

Comment:
Break lines after 80 chars? It might not be doable for constant strings, but pls find out.